src/mongo/base/DriverBase.class.php

```diff
@@ -183,22 +183,43 @@ protected function fetchGraph($query, $type, $collection=null,$includeProperties
 
         $ttlExpiredResources = false;
         $cursor->batchSize($cursorSize);
-        foreach($cursor as $result)
-        {
-            // handle MONGO_VIEWS that have expired due to ttl. These are expired
-            // on read (lazily) rather than on write
-            if ($type==MONGO_VIEW && array_key_exists(_EXPIRES,$result['value']))
-            {
-                // if expires < current date, regenerate view..
-                $currentDate = new \MongoDate();
-                if ($result['value'][_EXPIRES]<$currentDate)
+
+        $retries = 1;
+        $exception = null;
+        $cursorSuccess = false;
+
+        do {
+            try {
+                foreach($cursor as $result)
                 {
-                    // regenerate!
-                    $this->generateView($result['_id']['type'],$result['_id']['r']);
+                    // handle MONGO_VIEWS that have expired due to ttl. These are expired
+                    // on read (lazily) rather than on write
+                    if ($type==MONGO_VIEW && array_key_exists(_EXPIRES,$result['value']))
+                    {
+                        // if expires < current date, regenerate view..
+                        $currentDate = new \MongoDate();
+                        if ($result['value'][_EXPIRES]<$currentDate)
+                        {
+                            // regenerate!
+                            $this->generateView($result['_id']['type'],$result['_id']['r']);
+                        }
+                    }
+                    $graph->add_tripod_array($result);
                 }
+                $cursorSuccess = true;
+            } catch (\MongoCursorException $e) {
+                self::getLogger()->warn("MongoCursorException: " . $e->getMessage());
+                sleep(1);
+                $retries++;
+                $exception = $e;
             }
-            $graph->add_tripod_array($result);
+        } while ($retries < 30 && $cursorSuccess === false);
```

Comment:
seems reasonable.   was going to make the point that it makes sense to retry if the nature of the cursor exception is something you know you can recover from. but i dont think you can distinguish so a brute retry is probably all you can do here.